sdk/eventgrid/azure-eventgrid/azure/eventgrid/_consumer.py

```diff
@@ -20,31 +20,50 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-from ._models import DeserializedEvent
+from ._models import CloudEvent, EventGridEvent
 
 class EventGridConsumer(object):
     """
     A consumer responsible for deserializing event handler messages, to allow for access to strongly typed Event objects.
     """
+    def decode_cloud_event(self, cloud_event, **kwargs):
+        # type: (Union[str, dict, bytes], Any) -> CloudEvent
+        """Single event following CloudEvent schema will be parsed and returned as Deserialized Event.
+        :param cloud_event: The event to be deserialized.
+        :type cloud_event: Union[str, dict, bytes]
+        :rtype: CloudEvent
 
-    def deserialize_event(self, event, **kwargs):
-        # type: (Union[str, dict, bytes]) -> models.DeserializedEvent
-        """Single event following CloudEvent/EventGridEvent schema will be parsed and returned as DeserializedEvent.
-        :param event: The event to be deserialized.
-        :type event: Union[str, dict, bytes]
-        :keyword str encoding: The encoding that should be used. Defaults to 'utf-8'
-        :rtype: models.DeserializedEvent
-        :raises: :class:`ValueError`, when events do not follow CloudEvent or EventGridEvent schema.
+        :raise: :class:`ValueError`, when events do not follow CloudEvent schema.
         """
         encode = kwargs.pop('encoding', 'utf-8')
         try:
-            if isinstance(event, six.binary_type):
-                event = json.loads(event.decode(encode))
-            elif isinstance(event, six.string_types):
-                event = json.loads(event)
-            return DeserializedEvent(event)
+            cloud_event = CloudEvent._load(cloud_event, encode)
```

Comment:
Nice++  I would nitpick the naming to _from_json, but other than that this looks good.